File: examples/python/getting-started/minimal-example.py
from aws_durable_execution_sdk_python import (
    DurableContext,
    durable_execution,
    durable_step,
    StepContext,
)

@durable_step
def greet_user(step_context: StepContext, name: str) -> str:
    """Generate a greeting."""
    return f"Hello {name}!"

@durable_execution
def handler(event: dict, context: DurableContext) -> str:
    """Simple durable function."""
    name = event.get("name", "World")
    greeting = context.step(greet_user(name))
    return greeting

File: examples/java/getting-started/minimal-example.java
public class GreetingHandler extends DurableHandler<Map<String, String>, String> {

    @Override
    public String handleRequest(Map<String, String> event, DurableContext ctx) {
        var name = event.getOrDefault("name", "World");

        // Generate a greeting
        var greeting = ctx.step("greet-user", String.class, stepCtx -> {
            return "Hello " + name + "!";
        });

        return greeting;
    }
}
